src/twisted/internet/test/test_tcp.py

```diff
@@ -115,6 +115,28 @@ def connect(client, destination):
 
 
 
+class TCPClientTestsBuilder(ReactorBuilder):
+    """
+    Builder defining tests relating to L{IReactorTCP.connectTCP}.
+    """
+    def _freePort(self, interface='127.0.0.1'):
+        probe = socket.socket()
```

Comment:
There are other ways of probing for a free port that are already implemented as part of reactor builder tests, using port 0 with `listenTCP`.  We should use that technique instead.